packages/react/src/exchanges/hooks/useExchangeFilter.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  createExchangeFilter as createExchangeFilterAction,
+  getExchangeFilter,
+  getExchangeFilterError,
+  isExchangeFilterFetched,
+  isExchangeFilterLoading,
+  type StoreState,
+} from '@farfetch/blackout-redux';
+import { useCallback, useEffect, useMemo } from 'react';
+import { useSelector } from 'react-redux';
+import useAction from '../../helpers/useAction.js';
+import type { Config, PostExchangeFilterData } from '@farfetch/blackout-client';
+import type { UseExchangeFilterOptions } from './types/useExchangeFilter.js';
+
+/**
+ * Provides facilities to manage an exchange filter request.
+ */
+function useExchangeFilter(
+  exchangeFilterData?: PostExchangeFilterData,
+  options: UseExchangeFilterOptions = {},
+) {
+  const exchangeFilterDataHookParameter = exchangeFilterData;
+  const { enableAutoFetch = true, fetchConfig } = options;
+  const isLoading = useSelector((state: StoreState) =>
+    isExchangeFilterLoading(state),
+  );
+  const error = useSelector((state: StoreState) =>
+    getExchangeFilterError(state),
+  );
+  const exchangeFilterResult = useSelector((state: StoreState) =>
+    getExchangeFilter(state),
+  );
+  const isFetched = useSelector((state: StoreState) =>
+    isExchangeFilterFetched(state),
+  );
+  const createExchangeFilter = useAction(createExchangeFilterAction);
+
+  const create = useCallback(
+    (
+      exchangeFilterData:
+        | PostExchangeFilterData
+        | undefined = exchangeFilterDataHookParameter,
+      config: Config | undefined = fetchConfig,
+    ) => {
+      if (!exchangeFilterData) {
+        return Promise.reject(new Error('No data to filter provided'));
+      }
+
+      return createExchangeFilter(exchangeFilterData, config);
+    },
+    [exchangeFilterDataHookParameter, fetchConfig, createExchangeFilter],
+  );
+
+  useEffect(() => {
+    if (
+      !isLoading &&
+      !isFetched &&
+      enableAutoFetch &&
```

Comment:
If this effect will only create a exchange filter, then this should be called `enableAutoCreate`. But then the hook itself should be called `useCreateExchangeFilter` and the `isFetched` value should be changed to something like `isCreated`. I think this needs to be thought out a little bit more. Let's talk tomorrow about this.

Reply:
And done. I think I've changed everything we talked yesterday, but it you notice something missing, please let me know.